core/templates/pages/teach-page/teach-page.component.spec.ts

```diff
@@ -15,112 +15,193 @@
-    ctrl.$onInit();
-    expect(ctrl.activeTabName).toBe('teach');
+  it('should click on teach tab', () => {
+    component.ngOnInit();
+    expect(component.activeTabName).toBe('teach');
 
-    ctrl.onTabClick('teach');
+    component.onTabClick('teach');
 
-    // Function setTimeout is being used here in order to wait onhashchange
-    // event to finish. setTimeout is executed only after call stack is empty.
-    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
-    setTimeout(function() {
-      expect(windowRef.nativeWindow.location.hash).toBe('#teach');
-      expect(ctrl.activeTabName).toBe('teach');
-      done();
-    });
+    expect(windowRef.nativeWindow.location.hash).toBe('#teach');
+    expect(component.activeTabName).toBe('teach');
   });
 
-  it('should click on participation tab', function(done) {
-    ctrl.$onInit();
-    expect(ctrl.activeTabName).toBe('teach');
-
-    ctrl.onTabClick('participation');
+  it('should click on participation tab', (() => {
+    component.ngOnInit();
+    expect(component.activeTabName).toBe('teach');
 
-    // Function setTimeout is being used here in order to wait onhashchange
-    // event to finish. setTimeout is executed only after call stack is empty.
-    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
-    setTimeout(function() {
-      expect(windowRef.nativeWindow.location.hash).toBe('#participation');
-      expect(ctrl.activeTabName).toBe('participation');
-      done();
-    });
-  });
+    component.onTabClick('participation');
+    expect(windowRef.nativeWindow.location.hash).toBe('#participation');
+    expect(component.activeTabName).toBe('participation');
+  }));
 
-  it('should activate teach tab on init', function() {
+  it('should activate teach tab on init', () => {
     windowRef.nativeWindow.location.hash = '#teach';
 
-    ctrl.$onInit();
+    component.ngOnInit();
 
     expect(windowRef.nativeWindow.location.hash).toBe('#teach');
-    expect(ctrl.activeTabName).toBe('teach');
+    expect(component.activeTabName).toBe('teach');
   });
 
-  it('should activate participation tab on init', function() {
+  it('should activate participation tab on init', () => {
     windowRef.nativeWindow.location.hash = '#participation';
 
-    ctrl.$onInit();
+    component.ngOnInit();
 
     expect(windowRef.nativeWindow.location.hash).toBe('#participation');
-    expect(ctrl.activeTabName).toBe('participation');
+    expect(component.activeTabName).toBe('participation');
   });
 
   it('should get static image url', function() {
-    expect(ctrl.getStaticImageUrl('/path/to/image')).toBe(
+    expect(component.getStaticImageUrl('/path/to/image')).toBe(
       '/assets/images/path/to/image');
   });
 
-  it('should apply to teach with oppia', function() {
-    var applyToTeachWithOppiaEventSpy = spyOn(
-      SiteAnalyticsService, 'registerApplyToTeachWithOppiaEvent')
+  it('should apply to teach with oppia', fakeAsync(() => {
```

Comment:
I don't think we need to include this test in fakeAsync -- could you please check?

Reply:
I need fakeAsync for tick, I'll try to see if I can eliminate that with using delays with macrotasks